### ISSUE TITLE
Difficulty with Canadian postal code

### ISSUE THREAD

**Issue by valianttry** [mozilla/donate.mozilla.org#914]
I searched for him in Stripe, no luck.

**I tried to donate to you but your system says my Canadian postal code is not valid.  It IS valid.**

I replied, asked him to confirm he was entering the correct postal code associated with the card. He replied:
**Yes it is. It is L6K 0A4 I tried it with a space and without a space (Between the K and the 0)**

I just want to make sure it is indeed user error, and there's nothing wrong with the form. (Which is what i suspect.)

**Comment by valianttry** [comment]
I mean, plenty of Canadians tested this form :maple_leaf:

**Comment by ScottDowne** [comment]
TLDR We should update the postal code error message.

Could be a valid postal code, in that he lived there, maybe moved, or otherwise not the postal code verified with the card? I think this is also the case with https://github.com/mozilla/donate.mozilla.org/issues/902

I took a closer look at the error message, and it's not great. We show them "Invalid zip/postal code", The Stripe description for the error is "The card's zip code failed validation." I wonder if we can improve on these messages? Invalid/validation could mean a bunch of things to people. Something along the lines "The code didn't match the card"

It is fact that this isn't us validating the postal code.

I personally had a frustrating moment when my bank updated some of my info when they found out I moved based on opening another account, they updated my card info, didn't tell me, and I had a hell of a time next time I tried to use my card with my parents postal code.

**Comment by ScottDowne** [comment]
Maybe something like "Zip/postal code check failed" maybe also add in a bit along the lines of "Are you sure this is your current zip/postal code?"

**Comment by jbuck** [comment]
We rely on Stripe checkout now, so closing
